src/KeyStoreUtils.ts

```diff
@@ -57,8 +59,16 @@ export namespace KeyStoreUtils {
             if (!bip39.validateMnemonic(mnemonic)) { throw new Error('The given mnemonic could not be validated.'); }
         }
 
-        const seed = (await bip39.mnemonicToSeed(mnemonic, password)).slice(0, 32);
-        const keys = await generateKeys(seed);
+        let keys: { secretKey: Buffer, publicKey: Buffer };
+        const seed = await bip39.mnemonicToSeed(mnemonic, password);
+        if (derivationPath !== undefined && derivationPath.length > 0) {
+            const p = Ed25519.derivePath(derivationPath, seed.toString("hex"));
```

Comment:
Could we make this variable name more descriptive?

Reply:
P, for Path!

Reply:
new commit coming...